org.mixedrealitytoolkit.input/Readers/PinchInputReader.cs

```diff
@@ -145,16 +141,14 @@ protected virtual void OnDisable()
         /// </summary>
         protected virtual void Update()
         {
-            if (m_firstUpdate)
-            {
-                UpdateActionValidCaches();
-                m_firstUpdate = false;                
-            }
+            m_isTrackingStatePolyfilled = trackedPoseDriver.GetIsPolyfillDevicePose();
+            m_isSelectPolyfilled = !selectAction.action.HasAnyControls();
```

Comment:
From what I recall...I was noticed a performance hit by calling HasAnyControls every frame.   Hence the caching being added.  

I don't think that HasAnyControls() can change often (if at all), so probably should still be caching this

Reply:
> and I think there's an allocation in here somewhere

Only on the initial read, and we're good after that:

> After the first read, no further GC allocations should occur except if the set of controls is changed (e.g. by changing the binding mask or by adding/removing devices to/from the system).

https://docs.unity3d.com/Packages/com.unity.inputsystem@1.10/api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_controls:~:text=After%20the%20first%20read%2C%20no%20further%20GC%20allocations%20should%20occur%20except%20if%20the%20set%20of%20controls%20is%20changed%20(e.g.%20by%20changing%20the%20binding%20mask%20or%20by%20adding/removing%20devices%20to/from%20the%20system).

> I don't think that HasAnyControls() can change often (if at all)

This is the root of the bug this PR is fixing. Controls can change as input sources are detected, connected, removed, etc. The linked doc above confirms this:

> except if the set of controls is changed (e.g. by changing the binding mask or by adding/removing devices to/from the system)

I went ahead and profiled in-editor on WMR desktop, and the `HasAnyControls()` method looks pretty cheap (0.01ms and no allocations, split across 8 calls (2 for each of the 4 in the scene) in the `PinchInputReader`):
![image](https://github.com/user-attachments/assets/525302c5-d7b7-40b3-981b-f7da53e46f0a)

Reply:
I'm don't understand Unity's documentation with what I observe in the code.

HasAnyControls is this...
```cs
        public static bool HasAnyControls(this InputAction action)
        {
            return action?.controls.Count > 0;
        }
```

which always calls controls::get...
```cs
        public ReadOnlyArray<InputControl> controls
        {
            get
            {
                var map = GetOrCreateActionMap();
                map.ResolveBindingsIfNecessary();
                return map.GetControlsForSingleAction(this);
            }
        }
```

which always calls InputActionMap::GetControlsForSingleAction...
```cs
        internal ReadOnlyArray<InputControl> GetControlsForSingleAction(InputAction action)
        {
            Debug.Assert(m_State != null);
            Debug.Assert(m_MapIndexInState != InputActionState.kInvalidIndex);
            Debug.Assert(m_Actions != null);
            Debug.Assert(action != null);
            Debug.Assert(action.m_ActionMap == this);
            Debug.Assert(!action.isSingletonAction || m_SingletonAction == action);

            if (!controlsForEachActionInitialized)
                SetUpPerActionControlAndBindingArrays();

            return new ReadOnlyArray<InputControl>(m_ControlsForEachAction, action.m_ControlStartIndex,
                action.m_ControlCount);
        }
```

which always calls `new ReadOnlyArray`....

Is this `new` not really allocating memory????

Reply:
Ahhh ok....ReadOnlyArray is a struct.....    doh

Reply:
Ohh, good find! I didn't dig as deep into the code itself. Thank you for the analysis!